src/view.c

```diff
@@ -1792,6 +1792,58 @@ view_snap_to_region(struct view *view, struct region *region,
 	view_apply_region_geometry(view);
 }
 
+struct output *
+view_find_output(struct view *view, const char *to)
+{
+	assert(view);
+	struct output *new_output;
+	if (!strcasecmp(to, "left")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_LEFT);
+	} else if (!strcasecmp(to, "right")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_RIGHT);
+	} else if (!strcasecmp(to, "up")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_UP);
+	} else if (!strcasecmp(to, "down")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_DOWN);
+	} else {
+		new_output = output_from_name(view->server, to);
+	}
+	return new_output;
+}
+
+void
+view_move_to_output(struct view *view, struct output *output)
+{
+	assert(view);
+	if (view->fullscreen) {
+		return;
+	}
+
+	view_invalidate_last_layout_geometry(view);
+	view_set_output(view, output);
+	if (view_is_floating(view)) {
+		struct wlr_box geo = output_usable_area_in_layout_coords(output);
+		geo.width = MIN(view->pending.width, geo.width);
+		geo.height = MIN(view->pending.height, geo.height);
```

Comment:
I'm not convinced the size should be clipped when relocating outputs. Unlike simple moves, resizing can be potentially "destructive" in the sense that it could cause the window to alter its layout and possibly take other actions.

This might be better served by a new action like "FitToOutput" that will clip the width and height, allowing you to chain "MoveToOutput" and "FitToOutput" if you really want.

Reply:
thanks, I went to splitting into move and fit and also applied all other suggestions. This turned out pretty good since now I was able to re-use `view_place_initial`.